src/helpers/namedSeriesAll.ts

```diff
@@ -0,0 +1,32 @@
+import { isFunction, isPromise } from '../validators'
+
+type NamedSeriesResult<T extends Record<string, unknown>> = {
+  [K in keyof T]: T[K] extends Promise<infer U>
+    ? U
+    : T[K] extends () => Promise<infer U>
+    ? U
+    : never
+} & {}
```

Comment:
this `& {}` is to enforce eager type evaluation.

Without it:

```ts
```ts
namedSeriesAll({
  a: Promise.resolve(true),
  b: Promise.resolve(42),
})
// Promise<NamedSeriesResult<{
//   a: Promise<boolean>;
//   b: Promise<number>;
// }>>

With `& {}`:
```ts
// Promise<{
//   a: boolean;
//   b: number;
// }>
```